File: C++/Fibonacci SubProblems/no_of_ways_to_tile.cpp
#include<bits/stdc++.h>
using namespace std;

int tile(int n){
    int a[n+1];

    for(int i=0; i<=n; i++){
        a[i] = 0;
    }
    for(int i=0; i<=n; i++){
        if(i<4){
            a[i] = 1;
        } else if(i==4) {
            a[i] = 2;
        } else {
            a[i] = a[i-1] + a[i-4];
        }
    }
    return a[n];
}

int main(){
    int test, n;
    cin>>test;
    while(test--){
        cin>>n;
        cout<<tile(n)<<'\n';
    }
}

File: Python/Fibonacci SubProblems/no_of_ways_to_tile.py
# n*n floor with 1*4 tile
def tile(n):
    a = [0 for _ in range(n+1)]
    for i in range(1, n+1):
        if(i<=3):
            a[i] = 1
        elif(i==4):
            a[i] = 2
        else:
            a[i] = a[i-1] + a[i-4]
    return a[n]


test = int(input())
for i in range(test):
    n = int(input())
    print(tile(n))